packages/design-system/src/components/slider/types.ts

```diff
@@ -0,0 +1,109 @@
+import type {
+  NumberFieldProps as AriaNumberFieldProps,
+  SliderRenderProps as RACSliderRenderProps,
+  SliderProps as RACSliderProps,
+  SliderThumbProps as RACSliderThumbProps,
+  SliderThumbRenderProps as RACSliderThumbRenderProps,
+  SliderTrackRenderProps as RACSliderTrackRenderProps,
+  SliderTrackProps as RACSliderTrackProps,
+  SliderOutputProps as RACSliderOutputProps,
+} from 'react-aria-components';
+import type { RenderPropsChildren } from '../../types';
+import type { PartialDeep } from 'type-fest';
+import type { InputClassNames, InputRenderProps } from '../input';
+import type { GroupClassNames } from '../group';
+
+export type SliderClassNames = PartialDeep<{
+  slider: {
+    container: string;
+    label: string;
+    min: string;
+    max: string;
+  };
+  track: {
+    container: string;
+    track: string; // TODO: adjust the styles to use this class
+  };
+  thumb: {
+    container: string;
+    thumb: string;
+  };
+  input: InputClassNames;
+  output: {
+    container: string;
+  };
+  group: GroupClassNames;
+}>;
+
+export type LabelAlignment = 'top' | 'left';
+
+export type SliderRenderProps = RACSliderRenderProps;
+export type SliderThumbRenderProps = RACSliderThumbRenderProps;
+export type SliderTrackRenderProps = RACSliderTrackRenderProps;
+
+type BaseSliderProps = {
+  children?: RenderPropsChildren<SliderRenderProps>;
+  label?: string;
+  classNames?: SliderClassNames;
+  alignLabel?: LabelAlignment;
+  includeTextField?: boolean;
+  includeRangeLabel?: boolean;
+  includeOutputField?: boolean;
+  minValue?: number;
+  maxValue?: number;
+};
+
+export type SliderState = Omit<RACSliderRenderProps, 'state'> &
+  Required<Pick<BaseSliderProps, 'alignLabel'>>;
+
+export type SliderProps = Omit<
+  RACSliderProps,
+  'children' | 'classname' | 'style'
+> &
+  BaseSliderProps;
+
+type BaseSliderThumbProps = {
+  children?: RenderPropsChildren<SliderRenderProps>;
```

Comment:
I don't think you need to redefine `children` here unless you're actually changing the type (which I don't think you are). And then you could simplify why getting rid of the BaseSlideThumbProps and just add the classNames prop directly to the join

Reply:
This applies below too